src/protocol/sort/mod.rs

```diff
@@ -4,22 +4,55 @@ use std::fmt::Debug;
 mod apply;
 pub mod bit_permutations;
 mod compose;
+mod generate_sort_permutations;
 pub mod reshare;
 mod secureapplyinv;
 mod shuffle;
 
 #[allow(clippy::module_name_repetitions)]
 #[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
 pub enum SortStep {
-    BitPermutations,
+    ModulusConversion,
+    BitPermutation(u8),
+    ApplyInv(u8),
+    ComposeStep(u8),
 }
 
 impl Step for SortStep {}
 
 impl AsRef<str> for SortStep {
     fn as_ref(&self) -> &str {
+        const BITPERMUTATIONS: [&str; 64] = [
```

Comment:
This is quite ugly and repetitive. Since records are used in lower layers, I just created these steps per bit. Happy to incorporate a better way of doing this.

Reply:
macro?